### ISSUE TITLE
Check for updates

### ISSUE THREAD

**Issue by Hyde46** [Hyde46/hoard#40]
On start check for availble updates.
Compare running version of `hoard` with latest tag on github repository
Inform user of available update

**Comment by gautamprikshit1** [comment]
Can you provide a strategy on how to implement it. Thanks

**Comment by gautamprikshit1** [comment]
Can you help me with the implementation of a function for updating hoard cli?

**Comment by Hyde46** [comment]
Sure!
Similar to how trove files are imported by url [here](https://github.com/Hyde46/hoard/blob/main/src/hoard.rs#L583). You can create another function to request the latest version of `hoard`.
In bash, it could work like this:
```
  curl --silent "https://api.github.com/repos/$1/releases/latest" | # Get latest release from GitHub api
    grep '"tag_name":' |                                            # Get tag line
    sed -E 's/.*"([^"]+)".*/\1/'                                    # Pluck JSON value
```
translate that to a `reqwest` call and some string operations to find out the latest version tag of the project, then compare to the version of the currently running process
```
const VERSION: &str = env!("CARGO_PKG_VERSION");
```
and inform the user if there is a newer version available.
If there's no network connection, or any other issue with requesting the tag, fail silently.

Pluck the new function in somewhere around [here](https://github.com/Hyde46/hoard/blob/main/src/hoard.rs#L47)

**Comment by gautamprikshit1** [comment]
So I don't have to fetch the latest version package and replace the existing one

**Comment by Hyde46** [comment]
Initially no. 
Only informing the user about a new version would satisfy this feature. 
In an additional step, we could think about how self-update could work

**Comment by Hyde46** [comment]
Implemented with #238
